cobra/preprocessing/preprocessor.py

```diff
@@ -223,28 +223,87 @@ def from_pipeline(cls, pipeline: dict):
+        id_col_name : str
+            column name of the id column, can be None
+        target_column_name : str
+            column name of the target column
+
+        Returns
+        -------
+        tuple
+            tuple containing 2 lists of column names. (continuous_vars, discrete_vars)
+        """        
+        if id_col_name == None:
+            log.warning("id_col_name is equal to None. If there is no id column ignore this warning")
+        
+        # find continuous_vars and discrete_vars in the dateframe
+        col_dtypes = df.dtypes
+        discrete_vars = [col for col in col_dtypes[col_dtypes==object].index.tolist() if col not in [id_col_name, target_column_name]]
+        
+
+        for col in df.columns:
+            if col not in discrete_vars and col not in [id_col_name, target_column_name]: # omit discrete because a string, and target
+                val_counts = df[col].nunique()
+                if val_counts > 1 and val_counts <= 10: # the column contains less than 10 different values
+                    discrete_vars.append(col)
+
+        continuous_vars = list(set(df.columns)
+                            - set(discrete_vars) 
+                            - set([id_col_name, target_column_name]))
+        log.warning(
+            f"""Cobra automaticaly assumes that following variables are 
+            discrete: {discrete_vars}
+            continuous: {continuous_vars}
+            If you want to change this behaviour you can specify the discrete/continuous variables yourself with the continuous_vars and discrete_vars keywords. \nIt assumes that numerical comumns with less than or equal to 10 different values are categorical"""
+            )
+        return continuous_vars, discrete_vars
 
     def fit(
         self,
         train_data: pd.DataFrame,
         continuous_vars: list,
         discrete_vars: list,
         target_column_name: str,
+        id_col_name: str = None
     ):
         """Fit the data to the preprocessing pipeline.
+        If you put continious_vars and target_vars equal to `None` and give the id_col_name Cobra will guess which varaibles are continious and which are not 
 
         Parameters
         ----------
         train_data : pd.DataFrame
             Data to be preprocessed.
-        continuous_vars : list
-            List of continuous variables.
-        discrete_vars : list
-            List of discrete variables.
+        continuous_vars : list | None
+            List of continuous variables, can be None.
+        discrete_vars : list | None
+            List of discrete variables, can be None.
         target_column_name : str
             Column name of the target.
+        id_col_name : str, optional
+            _description_, by default None
         """
+        if not (continuous_vars and discrete_vars):
+            continuous_vars, discrete_vars = self.get_continous_and_discreate_columns(
```

Comment:
Typos in code can happen, but please always double-check your method names for typos, since this snowballs to the end user eventually.
`get_continuous_and_discrete_columns(...)`